### PR TITLE
Remove privilege flag from vpn-seed-server

### PR DESCRIPTION
**Pull request by ScheererJ**
**How to categorize this PR?**

/area networking
/kind cleanup

**What this PR does / why we need it**:
Remove privilege flag from vpn-seed-server.

It is not necessary for the vpn-seed-server to run as privileged container. The tun
device required for vpn operation can be mounted into the container namespace.
In case the container should run as non-root user in the future, it might help to
create a separate tun device in a privileged init container as described in
https://community.openvpn.net/openvpn/wiki/UnprivilegedUser#TUNTAPDevice .

**Which issue(s) this PR fixes**:
Partially addresses https://github.com/gardener/vpn/issues/41.

**Special notes for your reviewer**:

**Release note**:

```other operator
The vpn-seed-server/vpn-seed-server container no longer runs in privileged mode.
```

### CODE REVIEW

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Review by ialidzhikov — COMMENTED** [review]
Thanks for the PR.

I was wondering why it needs to mount `/dev/net/tun` from the host? Aren't there any side-effects, I mean what happens when 2 different `vpn-seed-server` Pods running on the same Node mount the same `/dev/net/tun`? I am not openvpn expert but my feeling is that it does not need to mount the `/dev/net/tun` from the host. Can such device be created for the container itself?

**Comment by ScheererJ** [comment]
> I was wondering why it needs to mount `/dev/net/tun` from the host? Aren't there any side-effects, I mean what happens when 2 different Pods running on the same Node mount the same `/dev/net/tun`? I am not openvpn expert but my feeling is that it does not need to mount the `/dev/net/tun` from the host. Can such device be created for the container itself?

That happens already now. `/dev/net/tun` is used to create another device, which is then used for the communication. You can also have a look at the log of the `vpn-seed-server` to confirm that another device was created.

Anyway, this is exactly the same setup we already run, i.e. all `vpn-seed-server` pods reused the same `/dev/net/tun` device as they shared the same `/dev` tree due to the `privileged` mode.

It is possible to create another device in an init container, but I do not see any benefits. For further details, you can take a look at https://www.kernel.org/doc/html/v5.10/networking/tuntap.html .

File: pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go
Comment (ialidzhikov) [inline pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go]:
```suggestion
	volumeNameDevNetTun       = "dev-net-tun"
```

to make the naming identical with the other volume names.

File: pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go
Comment (ialidzhikov) [inline pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go]: I would rather remove it completely as the default is `false` (unprivileged mode), no need to explicitly specify it.

```suggestion
```

File: pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go
Comment (ialidzhikov) [inline pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go]: Nit: I would rather vote to use

```diff
--- a/pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go
+++ b/pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go
@@ -195,6 +195,8 @@ func (v *vpnSeedServer) Deploy(ctx context.Context) error {
                vpa             = v.emptyVPA()
                igwSelectors    = v.getIngressGatewaySelectors()

+               hostPathCharDev = corev1.HostPathCharDev
+
                vpaUpdateMode    = vpaautoscalingv1.UpdateModeAuto
                controlledValues = vpaautoscalingv1.ContainerControlledValuesRequestsOnly
        )
@@ -478,7 +480,7 @@ func (v *vpnSeedServer) Deploy(ctx context.Context) error {
                                                        VolumeSource: corev1.VolumeSource{
                                                                HostPath: &corev1.HostPathVolumeSource{
                                                                        Path: volumeMountPathDevNetTun,
-                                                                       Type: (*corev1.HostPathType)(pointer.String(string(corev1.HostPathCharDev))),
+                                                                       Type: &hostPathCharDev,
                                                                },
                                                        },
                                                },
```

File: pkg/operation/botanist/component/vpnseedserver/vpn_seed_server_test.go
Comment (ialidzhikov) [inline pkg/operation/botanist/component/vpnseedserver/vpn_seed_server_test.go]:
```suggestion
											Type: &hostPathCharDev
```

**Review by ialidzhikov — COMMENTED** [review]
Okay, if you say that it is not causing a problem all vpn-seed-servers to reuse the same `/dev/net/tun` devices from the host, then this PR is fine for me.

@marwinski can you also review this PR?

I added some cosmetic suggestions only.

File: pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go
Comment (ScheererJ) [reply pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go]: Adapted accordingly in latest commit.

File: pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go
Comment (ScheererJ) [reply pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go]: Adapted accordingly in latest commit.

File: pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go
Comment (ScheererJ) [reply pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go]: Adapted accordingly in latest commit.

File: pkg/operation/botanist/component/vpnseedserver/vpn_seed_server_test.go
Comment (ScheererJ) [reply pkg/operation/botanist/component/vpnseedserver/vpn_seed_server_test.go]: Adapted accordingly in latest commit.

**Comment by ialidzhikov** [comment]
lgtm, I am waiting for review from @marwinski as he was involved in https://github.com/gardener/vpn/issues/41. I expect the same review for https://github.com/gardener/gardener/pull/6352.

**Review by marwinski — APPROVED** [review]

**Comment by rfranzke** [comment]
> @marwinski: adding LGTM is restricted to approvers and reviewers in OWNERS files.

/approve

**Pull request merged**
